lib/spack/spack/package.py

```diff
@@ -1217,6 +1236,15 @@ def rpath_args(self):
         return " ".join("-Wl,-rpath,%s" % p for p in self.rpath)
 
 
+class PythonExtension(Package):
+    def setup_dependent_environment(self, env, dependent_spec):
+        pass
+
+    def setup_environment(self, env):
+        site_packages = glob.glob(join_path(self.spec.prefix.lib, "python*/site-packages"))
+        if site_packages:
+            env.prepend_path('PYTHONPATH', site_packages[0])
```

Comment:
@tgamblin I tried to solve the issue on `PYTHONPATH` with a base class that just implements the two setup functions for python extensions (an example of use is in `py-nose`). If this is fine with you I'll proceed modifying the rest of the packages.